database/database-orders.go

```diff
@@ -41,21 +45,21 @@ func (s *Service) AddOrder(order models.Order, userID uuid.UUID, currentDate tim
 // Limit is how many items. Offset is from where to where the data is used
 func (s *Service) GetOrders(orderDetails models.Order, limit int, offset int) (*[]models.Order, error) {
 	var orders *[]models.Order
-	if err := s.DB.Where(models.Order{UserID: orderDetails.UserID, Type: orderDetails.Type, Symbol: orderDetails.Symbol}).Limit(limit).Offset(offset).Error; err != nil {
+	if err := s.DB.Where(models.Order{UserID: orderDetails.UserID, Type: orderDetails.Type, Symbol: orderDetails.Symbol}).Limit(limit).Offset(offset).Find(&orders).Error; err != nil {
 		return nil, err
 	}
 
 	return orders, nil
 }
 
 // GetOrderByID gets an order by the orderID
-func (s *Service) GetOrderByID(orderID uuid.UUID) (*models.Order, error) {
-	var order *models.Order
+func (s *Service) GetOrderByID(orderID uuid.UUID) (models.Order, error) {
+	var order []models.Order
 	if err := s.DB.Where(models.Order{OrderID: orderID}).First(&order).Error; err != nil {
-		return nil, err
+		return models.Order{}, err
 	}
 
-	return order, nil
+	return order[0], nil
```

Comment:
I think we should actually just return order and have order be of type `models.Order`. `.First` only finds a single element.

Reply:
will update it in the next commit